compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1849,8 +1849,7 @@ private boolean checkNextFuncReturnType(BType returnType) {
             return false;
         }
 
-        List<BType> types = getAllTypes(returnType);
-        types.removeIf(type -> type.tag == TypeTags.NEVER);
+        List<BType> types = new ArrayList<>(((BUnionType) returnType).getMemberTypes());
```

Comment:
Why change `getAllTypes(returnType);` to new `ArrayList<>(((BUnionType) returnType).getMemberTypes());`